src/print.c

```diff
@@ -260,7 +260,25 @@ print_file (int fd,
   g_autoptr(GUnixOutputStream) ostream = NULL;
   int fd2;
 
-  title = g_strdup_printf ("Document from %s", app_id);
+  g_autofree char *app_desktop_fullname = NULL;
+  if (!g_str_has_suffix (app_id, ".desktop"))
+    {
+      app_desktop_fullname = g_strconcat (app_id, ".desktop", NULL);
+    }
+  else
+    {
+      app_desktop_fullname = g_strdup (app_id);
+    }
+
+  g_autoptr (GDesktopAppInfo) app_info = g_desktop_app_info_new (app_desktop_fullname);
+  // ensures the app_name won't be NULL even if the app_id.desktop file doesn't exist
+  g_autofree char *app_name = g_strdup (app_id);
```

Comment:
same for app_info and app_name. Please, move the var declarations to the top alongside the others.